packages/experiments/src/components/Button/Button.types.tsx

```diff
@@ -9,16 +16,28 @@ export type IButtonStates = 'baseState' | 'enabled' | 'disabled' | 'expanded';
 
 export type IButtonVariants = 'baseVariant' | 'primary' | 'circular';
 
-export type IButtonSlots = 'root' | 'stack' | 'text' | 'icon' | 'menuIcon';
+export interface IButtonSlots {
+  // TODO: remove test slots
+  // TODO: make sure children are still disallowed via typings when removed here
+  root?: ISlotProp<React.ButtonHTMLAttributes<any>>;
+  stack?: ISlotProp<IHorizontalStackProps>;
+  content?: ISlotProp<ITextProps, IChildrenProp>;
+  icon?: ISlotProp<IIconProps, IIconDefaultProp>;
+  menu?: ISlotProp<IContextualMenuProps>;
+  menuIcon?: ISlotProp<IIconProps, IIconDefaultProp>;
+  test1?: ISlotProp<ITextProps, IChildrenProp>;
+  test2?: ISlotProp<React.AllHTMLAttributes<any>>;
+}
 
 export interface IButton {}
 
-export interface IButtonProps extends IStyleableComponentProps<IButtonProps, IButtonStyles> {
-  as?: keyof JSX.IntrinsicElements;
+export interface IButtonProps extends IButtonSlots, IStyleableComponentProps<IButtonProps, IButtonStyles> {
```

Comment:
this is great that these are ACTUAL props, not tons of things used to customize the internals of the control.